From 8b0073dcde59ae9420099dd7e97946c99078a14f Mon Sep 17 00:00:00 2001
From: Damien Le Moal <dlemoal@kernel.org>
Date: Mon, 14 Oct 2024 17:32:51 +0900
Subject: [PATCH 02/21] arm64: dts: rockchip: Add rockpro64 overlay for PCIe
 Gen2

Add a DTS overlay file for the rk3399-rockpro64 DT for increasing the
PCIe3x4 slot maximum link speed to Gen2.

Signed-off-by: Damien Le Moal <dlemoal@kernel.org>
---
 arch/arm64/boot/dts/rockchip/Makefile             |  1 +
 .../dts/rockchip/rk3399-rockpro64-pcie-gen2.dtso  | 15 +++++++++++++++
 2 files changed, 16 insertions(+)
 create mode 100644 arch/arm64/boot/dts/rockchip/rk3399-rockpro64-pcie-gen2.dtso

diff --git a/arch/arm64/boot/dts/rockchip/Makefile b/arch/arm64/boot/dts/rockchip/Makefile
index d49f8285c394..a707bd7595be 100644
--- a/arch/arm64/boot/dts/rockchip/Makefile
+++ b/arch/arm64/boot/dts/rockchip/Makefile
@@ -75,6 +75,7 @@ dtb-$(CONFIG_ARCH_ROCKCHIP) += rk3399-rock960.dtb
 dtb-$(CONFIG_ARCH_ROCKCHIP) += rk3399-rockpro64-v2.dtb
 dtb-$(CONFIG_ARCH_ROCKCHIP) += rk3399-rockpro64.dtb
 dtb-$(CONFIG_ARCH_ROCKCHIP) += rk3399-rockpro64-pcie-ep.dtbo
+dtb-$(CONFIG_ARCH_ROCKCHIP) += rk3399-rockpro64-pcie-gen2.dtbo
 dtb-$(CONFIG_ARCH_ROCKCHIP) += rk3399-sapphire.dtb
 dtb-$(CONFIG_ARCH_ROCKCHIP) += rk3399-sapphire-excavator.dtb
 dtb-$(CONFIG_ARCH_ROCKCHIP) += rk3399pro-rock-pi-n10.dtb
diff --git a/arch/arm64/boot/dts/rockchip/rk3399-rockpro64-pcie-gen2.dtso b/arch/arm64/boot/dts/rockchip/rk3399-rockpro64-pcie-gen2.dtso
new file mode 100644
index 000000000000..834acdce44f4
--- /dev/null
+++ b/arch/arm64/boot/dts/rockchip/rk3399-rockpro64-pcie-gen2.dtso
@@ -0,0 +1,15 @@
+// SPDX-License-Identifier: (GPL-2.0+ OR MIT)
+/*
+ * DT-overlay to run the PCIe Dual Mode controller at Gen2 speed.
+ */
+
+/dts-v1/;
+/plugin/;
+
+&pcie0 {
+	max-link-speed = <2>;
+};
+
+&pcie0_ep {
+	max-link-speed = <2>;
+};
-- 
2.47.1

